posit-dev/positron
# Updates to use the editor font in the Data Grid

### Intent

> Describe briefly what problem this pull request resolves, or what new feature it introduces. Include screenshots of any new or altered UI. Link to any GitHub issues that are related.

Users want data values to be displayed in a fixed-width font in the `PositronDataExplorer`. This PR adds the capability to use the fixed-width editor font to display values in any `PositronDataGrid`. The `PositronDataExplorer` has been updated to use this capability.

Here's a screenshot:

![image](https://github.com/posit-dev/positron/assets/853239/91e9b246-11ea-4fb3-be58-fe1681ab6cd7)

This capability is enabled by setting `useEditorFont` to `true` in the constructor of a `DataGridInstance`:

<img width="622" alt="image" src="https://github.com/posit-dev/positron/assets/853239/1db87cb9-02e2-4a59-b511-db59bc81c9b5">

Users also want summary information to be displayed in a fixed width font in the `PositronDataExplorer`. This PR adjusts the `ProfileNumber` and `ProfileString` components to use the fixed-width editor font.

Here's a screenshot:

![image](https://github.com/posit-dev/positron/assets/853239/1da24355-3de2-4547-a634-88ce434d9949)

### Approach

> Describe the approach taken and the tradeoffs involved if non-obvious; add an overview of the solution if it's complicated.

I repurposed code from the `Console` component for this work and built a new utility function called `editorFontApplier` that can be used to apply the editor font to any `HTMLElement`. It's very simple to use:

```TypeScript
// Main useEffect.
useEffect(() => {
	// Create the disposable store for cleanup.
	const disposableStore = new DisposableStore();

	// Use the editor font.
	disposableStore.add(
		editorFontApplier(
			props.instance.configurationService,
			ref.current
		)
	);

	// Return the cleanup function that will dispose of the disposables.
	return () => disposableStore.dispose();
}, [props.instance.configurationService]);
```

### QA Notes

> Add additional information for QA on how to validate the change, paying special attention to the level of risk, adjacent areas that could be affected by the change, and any important contextual information not present in the linked issues.

You should see the editor font in the PositronDataExplorer. Also, you should be able to change the editor font and see it change automatically in any open `PositronDataExplorer`.
